custom_components/tahoma/coordinator.py

```diff
@@ -50,16 +55,36 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed(f"Error communicating with the TaHoma API: {exception}")
         else:
             for event in events:
+                if event.name == "DeviceAvailableEvent":
+                    self.devices[event.deviceurl].available = True
+
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
                         if state.name not in device.states:
                             device.states[state.name] = state
                         device.states[state.name].value = self._get_state(state)
+
+                if (
+                    event.name == "ExecutionRegisteredEvent"
+                    and event.exec_id in self.executions
+                ):
+                    self.update_interval = timedelta(seconds=1)
+
+                if (
+                    event.name == "ExecutionStateChangedEvent"
+                    and event.exec_id in self.executions
+                ) and event.new_state == "COMPLETED":
+                    self.executions.pop(event.exec_id, None)
```

Comment:
```suggestion
                    del self.executions[event.exec_id]
```

```
>>> a = {"a1": "v1"}
>>> del a["a1"]
>>> a
{}
```

Reply:
Indeed. Firstly I didn't check here `if event.exec_id in self.executions`, thus I could not use `del`. However since we check it up front, we could use `del()`.